ocp_resources/resource.py

```diff
@@ -964,41 +994,15 @@ def update_replace(self, resource_dict: Dict[str, Any]) -> None:
         hashed_resource_dict = self.hash_resource_dict(resource_dict=resource_dict)
         self.logger.info(f"Replace {self.kind} {self.name}: \n{hashed_resource_dict}")
         self.logger.debug(f"\n{yaml.dump(hashed_resource_dict)}")
-        self.api.replace(body=resource_dict, name=self.name, namespace=self.namespace)
-
-    @staticmethod
-    def retry_cluster_exceptions(
-        func,
-        exceptions_dict: Dict[type[Exception], List[str]] = DEFAULT_CLUSTER_RETRY_EXCEPTIONS,
-        timeout: int = TIMEOUT_10SEC,
-        sleep_time: int = 1,
-        **kwargs: Any,
-    ) -> Any:
-        try:
-            sampler = TimeoutSampler(
-                wait_timeout=timeout,
-                sleep=sleep_time,
-                func=func,
-                print_log=False,
-                exceptions_dict=exceptions_dict,
-                **kwargs,
-            )
-            for sample in sampler:
-                return sample
-
-        except TimeoutExpiredError as exp:
-            if exp.last_exp:
-                raise exp.last_exp
-
-            raise
+        with get_retry_cluster_exceptions_client(client=self.client, retry_count=5, backoff_factor=1) as self.client:
```

Comment:
`self.api` is a property so we cannot pass it a client but not sure using `as self.client` is correct, @myakove wdyt?